### PR TITLE
Added Circular array documentation

### LINKED ISSUE #1817: Add the docs for the circular array.
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md).

### Description

 This issue about the implemention and use of circular array, a datastructure that connects the last index back to the first, effectively creating a circular layout for storage. Key features include fixed size, allowing efficient space utilization by reusing indices after deletions, and support for constant-time access to elements via their index. Circular arrays facilitate efficient operations for queues and buffers, as elements can be enqueued and dequeued without shifting others, leading to reduced overhead. They also require careful management of two pointers (or indices) to track the front and rear positions, enabling seamless wrap-around behavior when reaching the end of the array.

### Self-service

- [X] I'd be willing to address this documentation request myself.

### PR DESCRIPTION
**Pull request by Richajaishwal0**
### Description
The circular array, or circular buffer, is a linear data structure that wraps around to the beginning once it reaches the end, allowing efficient use of memory by reusing space without reallocating the array. It operates with a fixed size and is commonly utilized in queue implementations and buffering systems. Key operations include initialization, insertion (enqueue), deletion (dequeue), and checking the full or empty status. The provided implementations in Python, Java, and C++ demonstrate how to manage the circular nature of the array while maintaining optimal performance for these operations. The time complexity for enqueueing and dequeueing is \(O(1)\), while the space complexity is \(O(n)\), where \(n\) is the fixed size of the array.

Fixes #1817

### Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Circular_array/Introduction.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
+---
+id: Circular Array introduction
```
Comment (ajay-dhangar) [inline docs/Circular_array/Introduction.md]: replace `id: Circular Array introduction` with `id: circular-array-introduction`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update only `id` in `docs/Circular_array/Introduction.md` file

**Comment by Richajaishwal0** [comment]
@ajay-dhangar please check once

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**